### PR TITLE
BFD-3521 Create generalized global CloudWatch alert on lambda failures Update alarms.tf

### PR DESCRIPTION
**Pull request by MahiFentaye**
BFD-3521 Create generalized global CloudWatch alert on lambda failures

**JIRA Ticket:**
BFD-3521

### What Does This PR Do?
Creates a global CloudWatch Alarm that is defined to send alerts to the BFD-Warnings/Alerts slack channel  that notify us of any reported Lambda failures
The Alert(s) are implemented in Terraform
The first alert from the Alarm(s) create a slack notification, but subsequent alerts do not cause any further action (the Incidents are not self-resolving)
This is simply to reduce VictorOps Incident spam, and the On-Call should be resolving these Incidents manually anyways
tl;dr: do not add an ok_action to the CW Alarm

Please follow standard Git commit message guidelines:
* First line should be a capitalized, short (50 chars or less) summary.
* The rest of the message should be in standard Markdown format, wrapped to 72 characters.
* Describe your changes in imperative mood, e.g. "make xyzzy do frotz" instead of "[This patch] makes xyzzy do frotz" or "[I] changed xyzzy to do frotz", as if you are giving orders to the codebase to change its behavior.
* List all relevant Jira issue keys, one per line at the end of the message, per: <https://confluence.atlassian.com/jirasoftwarecloud/processing-issues-with-smart-commits-788960027.html>.

Reference: <https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project>.
-->

Replace me.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [ ] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

Have you fully verified and tested these changes? Is the acceptance criteria met? Please provide reproducible testing instructions, code snippets, or screenshots as applicable.

### CODE REVIEW

File: ops/terraform/env/mgmt/alarms.tf [hunk 1]
```diff
@@ -106,3 +106,29 @@ resource "aws_cloudwatch_metric_alarm" "ec2_failing_instances" {
...
+
+  metric_name         = "Errors"
+  namespace           = "AWS/Lambda"
+  period              = 60
+  statistic           = "Sum"
+
+  #FIXME: replace when slack alert is in mgmt
+  alarm_actions       = [data.aws_sns_topic.internal_alert_slack.arn]
+
+}
```
Comment (malessi) [inline ops/terraform/env/mgmt/alarms.tf:134]: Just a small whitespace removal here!

```suggestion
}
```

File: ops/terraform/env/mgmt/alarms.tf [hunk 2]
```diff
@@ -106,3 +106,29 @@ resource "aws_cloudwatch_metric_alarm" "ec2_failing_instances" {
...
+
+resource "aws_cloudwatch_metric_alarm" "lambda_errors" {
+  alarm_name          = "bfd-mgmt-lambda-error"
+  comparison_operator = "GreaterThanThreshold"
+  datapoints_to_alarm = 1
+  evaluation_periods  = 1
+  threshold           = 0
+  actions_enabled = true
+  treat_missing_data  = "ignore"
```
Comment (malessi) [inline ops/terraform/env/mgmt/alarms.tf:122]: Just re-running `terraform fmt` here to align all the properties:

```suggestion
  actions_enabled     = true
  treat_missing_data  = "ignore"
```

**Review by malessi — CHANGES_REQUESTED** [review]
This looks great, thanks for working on this! I just have a couple _small_ formatting issues, and then this will be a ✅.

**Review by dondevun — APPROVED** [review]

**Review by malessi — APPROVED** [review]
This looks great! From the Slack notifications we got last night it seems this Alarm is already giving us some much needed awareness. Thanks Mahi!

**Review by aschey-forpeople — APPROVED** [review]

**Pull request merged**